### PR TITLE
Fixes #12939: Package management technique fails to check if a package is absent in the "latest" version

### PR DESCRIPTION
**Pull request by Fdall**
https://www.rudder-project.org/redmine/issues/12939

### CODE REVIEW

File: techniques/applications/packageManagement/1.1/packageManagement.st [hunk 1]
```diff
@@ -80,11 +80,14 @@ bundle agent package_management {
...
       "version_description[${index_pkg}]" string     => "in latest available version",
-                                          ifvarclass => "version_latest_${index_pkg}";
+                                          ifvarclass => "version_latest_${index_pkg}.state_present_${index_pkg}";
       "version_description[${index_pkg}]" string     => "in any version",
                                           ifvarclass => "!version_latest_${index_pkg}.!version_specified_${index_pkg}";
       "version_description[${index_pkg}]" string     => "in version ${version[${index_pkg}]}",
                                           ifvarclass => "version_specified_${index_pkg}";
+      # Non compatibility is check is done by the "package_state_options" this is just to inform the user
+      "version_description[${index_pkg}]" string     => "in latest available version is not supported and",
+                                          ifvarclass => "version_latest_${index_pkg}!state_present_${index_pkg}";
```
Comment (amousset) [inline techniques/applications/packageManagement/1.1/packageManagement.st]: missing operator

File: techniques/applications/packageManagement/1.1/packageManagement.st [hunk 2]
```diff
@@ -80,11 +80,14 @@ bundle agent package_management {
 
       # Version
       "version_description[${index_pkg}]" string     => "in latest available version",
-                                          ifvarclass => "version_latest_${index_pkg}";
+                                          ifvarclass => "version_latest_${index_pkg}.state_present_${index_pkg}";
```
Comment (amousset) [inline techniques/applications/packageManagement/1.1/packageManagement.st]: I think we have almost the same in ncf

**Comment by Fdall** [comment]
Commit modified

File: techniques/applications/packageManagement/1.0/packageManagement.st [hunk 3]
```diff
@@ -71,11 +71,14 @@ bundle agent package_management {
 
       # Version
       "version_description[${index_pkg}]" string     => "in latest available version",
-                                          ifvarclass => "version_latest_${index_pkg}";
+                                          ifvarclass => "version_latest_${index_pkg}.state_present_${index_pkg}";
       "version_description[${index_pkg}]" string     => "in any version",
                                           ifvarclass => "!version_latest_${index_pkg}.!version_specified_${index_pkg}";
       "version_description[${index_pkg}]" string     => "in version ${version[${index_pkg}]}",
                                           ifvarclass => "version_specified_${index_pkg}";
+      # Non compatibility is check is done by the "package_state_options" this is just to inform the user
```
Comment (amousset) [inline techniques/applications/packageManagement/1.0/packageManagement.st]: Non-compatibility check

**Comment by Fdall** [comment]
Commit modified

**Comment by Fdall** [comment]
OK, merging this PR

**Pull request merged**